drivers/i2s/i2s_esp32.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_I2S_I2S_ESP32_H_
+#define ZEPHYR_DRIVERS_I2S_I2S_ESP32_H_
+
+#include "esp_types.h"
+#include "esp_err.h"
+#include "soc/i2s_periph.h"
+#include "soc/rtc_periph.h"
+#include "hal/i2s_hal.h"
+
+struct queue_item {
+	void *buffer;
+	size_t size;
+};
+
+/* Minimal ring buffer implementation */
+struct ring_buffer {
```

Comment:
Why not use the Zephyr builtin ring-buffer library?

Reply:
I have updated the driver code to use Zephyr message-queue